src/sema/eval.rs

```diff
@@ -196,3 +196,361 @@ pub fn eval_const_rational(
         )),
     }
 }
+
+/// Function that recurses the expression, and calls a function that folds number literals in the expression: eval_constants_in_expression().
```

Comment:
```suggestion
/// Function that recurses the expression and folds number literals by calling 'eval_constants_in_expression'.
```

I finally understood what you meant. This is a suggestion to make your comment clearer.